mqtt_as/mqtt_as.py

```diff
@@ -429,12 +430,18 @@ def close(self):
             if sz != b"\x02":
                 raise OSError(-1)
             rcv_pid = await self._as_read(2)
-            self.dprint("received PUBACK", rcv_pid)
-            self.rcv_pids.add(rcv_pid[0] << 8 | rcv_pid[1])
+            pid = rcv_pid[0] << 8 | rcv_pid[1]
+            self.dprint("received PUBACK", pid)
+            if pid in self.rcv_pids:
+                self.rcv_pids.discard(pid)
+            else:
+                self.dprint("PUBACK unknown pid", pid)
+                raise OSError(-1)
```

Comment:
Discard received pids from rcv_pid instead of adding them. Makes it possible to detect unknown pids (although that should never happen) but removes the need for pruning if set is reinitialized on connect.